source/opt/folding_rules.cpp

```diff
@@ -1877,7 +1877,7 @@ FoldingRule FMixFeedingExtract() {
         inst->GetSingleWordInOperand(kExtractCompositeIdInIdx);
     Instruction* composite_inst = def_use_mgr->GetDef(composite_id);
 
-    if (composite_inst->opcode() != spv::Op::OpExtInst) {
+    if (!spvOpcodeIsExtInst(composite_inst->opcode())) {
```

Comment:
Why did you want to change every place we look for OpExtInst? In this case we are looking for FMix, which should never have a forward reference. I think you can be more selective in the place that you check for the new opcode.

Reply:
Idea was that OpExtInst & OpExtInstWithForward were the "same" instructions, but usage of the later was reserved in a subset of cases. So I wanted to encourage not having to care about whether it's OpExtInst or OpExtInstWithForward: we just want to know if it's an extended instruction. The legality of the opcode selection is the responsibility of other parts of SPIR-V tools.

Reply:
But I can be more selective yes.

Reply:
Reverted those cases, and renamed the function to something more truthful.